### PR TITLE
Added name of offending record to error message in SamPairUtil

### PR DESCRIPTION
**Pull request by yfarjoun**
This PR was inspired by https://github.com/broadinstitute/picard/issues/201 where the error doesn't allow the user to investigate the offending record. By adding the record name to the exception message, the user can investigate more easily and fix the input.

### CODE REVIEW

**Comment by yfarjoun** [comment]
Apologies about the CRLF->LF change. please use "hide whitespaces" to view the actual code change.

File: src/main/java/htsjdk/samtools/SamPairUtil.java [hunk 1]
```diff
@@ -1,513 +1,513 @@
...
+
+        /**
+         * Populate this.records if necessary.
+         */
+        private void advance() {
+
+            // No need to advance if we have records remaining
+            if (!records.isEmpty()) return;
+
+            /*
```
Comment (pshapiro4broad) [inline src/main/java/htsjdk/samtools/SamPairUtil.java]: It's good to fix this -- it's not a valid javadoc so it shouldn't start with ** -- but somehow the entire file is showing up as changed, as a whitespace change.

Can you make another PR that doesn't show every line as being different? Also, multi-line comments usually have a `*` character at the start of each line. Or, you could use `//` for each line of a multi-line comment.
